homework/hw9/task1_merge.py

```diff
@@ -0,0 +1,31 @@
+"""Task 1.
+
+Write a function that merges integer from sorted files and returns an iterator
+file1.txt:
+1
+3
+5
+file2.txt:
+2
+4
+6
+"""
+from itertools import zip_longest
+from pathlib import Path
+from typing import Iterator, List, Union
+
+
+def merge_sorted_files(file_list: List[Union[Path, str]]) -> Iterator:
+    """Merge sorted files.
+
+    Merges any amount of sorted files.
+
+    :param file_list: List with path/str to sorted files
+    """
+    files = [open(path) for path in file_list]
+    for tup in zip_longest(*files):
```

Comment:
In this task we should ensure that any sorted sequences in files will give us iterator with sorted values, e.g for files with 1 2 4 and 3 5 8 we have to achieve 1 2 3 4 5 8. Imagine that each file is too big and we don't have enough memory to store it. Condition about "... from sorted files ..." is a key part in task's text, which guarantee us that there will be no sequences like 1 3 5 2